doc/basics.md

```diff
@@ -53,6 +53,14 @@ rustup-toolchain-install-master -f -n master -c rustc-dev -c llvm-tools
 rustup override set master
 ```
 
+**Note:** You may get compiler errors when building Clippy, even if you
+didn't change anything. That's because, as mentioned earlier, Clippy
+depends on rustc internals. There is [discussion][nightly-build-issue]
+on ways to make sure that rustc does not break Clippy in the future
+so that Clippy always builds.
```

Comment:
Should we mention that this is called a tool breakage and link to forge docs?

Reply:
No, this is different from classic tool breakages.

Reply:
Hmm, I guess I didn't realize that. Are there docs that talk about the differences between classic tool breakages and clippy tool breakages?

Reply:
Yeah, see https://rustc-dev-guide.rust-lang.org/contributing.html#external-dependencies-subtree and the following section.

TL;DR: Clippy breakages will never appear in nightlies delivered with rustups, so they aren't user facing.

Reply:
So when there's an internal breaking change, clippy is updated in rust-lang/rust, but then has to be brought over to this repository (rust-lang/rust-clippy)?

Reply:
Yes, exactly.